sympy/matrices/expressions/dotproduct.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import print_function, division
+
+from sympy.core import Basic, sympify
+from sympy.matrices.expressions.matexpr import (MatrixExpr)
+
+
+class DotProduct(MatrixExpr):
+    """
+        Dot Product of vector matrices
+    """
+
+    def __new__(cls, *args):
+        args = list(map(sympify, args))
+        if validate(*args):
```

Comment:
I don't see the point of having a separate validation function. Why not just have the logic here `__new__`.